app/views/support/ABTest.scala

```diff
@@ -142,41 +140,45 @@ object RecurringPaymentTest extends TestTrait {
   )
 }
 
-case class ChosenVariants(variants: Seq[Variant]) {
-  def asJson = Json.toJson(variants)
-  def encodeURL = URLEncoder.encode(asJson.toString, StandardCharsets.UTF_8.name())
-}
-
 object Test {
+  private val MaxTestId = 100
+  val CookiePrefix     = "gu.contributions.test"
+  val TestIdCookieName = s"$CookiePrefix.id"
 
-  val allTests = List(AmountHighlightTest) // only used in unit tests
+  val allTests = List(AmountHighlightTest, PaymentMethodTest, RecurringPaymentTest)
 
-  def pickVariant[A](countryGroup: CountryGroup, request: Request[A], test: TestTrait): Variant = {
+  def cookieName(v: Variant) = s"$CookiePrefix.${v.testSlug}"
+  def cookieName(t: TestTrait) = s"$CookiePrefix.${t.slug}"
+
+  def testIdFor[A](request: Request[A]): Int = request.cookies.get(TestIdCookieName) map(_.value.toInt) getOrElse Random.nextInt(MaxTestId)
+  def testIdCookie(id: Int) = Cookie(TestIdCookieName, id.toString, maxAge = Some(604800))
+  def variantCookie(v: Variant) = Cookie(cookieName(v), v.variantSlug, maxAge = Some(604800))
 
+  def pickVariant[A](countryGroup: CountryGroup, request: Request[A], test: TestTrait): Variant = {
     def pickRandomly: Variant = {
       val n = Random.nextDouble
       test.variantRangesByCountry(countryGroup).dropWhile(_._1 < n).head._2
     }
 
     def pickByQueryStringOrCookie[A]: Option[Variant] = {
       val search: Option[String] = request.getQueryString(test.slug)
-        .orElse(request.cookies.get(test.slug + "_GIRAFFE_TEST").map(_.value))
-      test.variantsByCountry(countryGroup).find(_.variantSlug == search.getOrElse(None))
+        .orElse(request.cookies.get(s"$CookiePrefix.${test.slug}").map(_.value))
+      test.variantsByCountry(countryGroup).find(_.variantSlug == search.getOrElse(""))
     }
 
     pickByQueryStringOrCookie getOrElse pickRandomly
   }
 
-  def createCookie(variant: Variant): Cookie = {
-    Cookie(variant.testSlug+"_GIRAFFE_TEST", variant.variantSlug, maxAge = Some(604800))
+  def pickTest[A](request: Request[A], userId: Int): TestTrait = {
+    def pickByQueryString = allTests.find(t => request.getQueryString(t.slug).nonEmpty)
+    def pickByCookie = allTests.find(t => request.cookies.get(s"$CookiePrefix.${t.slug}").nonEmpty)
+    def pickRandomly = allTests(userId % allTests.length)
+
+    pickByQueryString orElse pickByCookie getOrElse pickRandomly
```

Comment:
Small details, but I don't think `pickByQueryStringOrCookie` needs a type parameter. Also, the code can be made even more readable (yay!) by using the `contains` method:

```scala
test.variantsByCountry(countryGroup).find(variant => search.contains(variant.variantSlug))
```